WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -140,7 +150,51 @@ private extension OrderDetailsViewController {
                                          action: #selector(editOrder))
         editButton.accessibilityIdentifier = "order-details-edit-button"
         editButton.isEnabled = viewModel.editButtonIsEnabled
-        navigationItem.rightBarButtonItem = editButton
+        navigationItem.rightBarButtonItems = [editButton] + orderNavigationRightBarButtonItems()
+    }
+
+    func orderNavigationRightBarButtonItems() -> [UIBarButtonItem] {
+        guard viewModels.count > 1 else { return [] }
+
+        let upArrowButon = UIBarButtonItem(
+            image: UIImage(systemName: "chevron.up"),
+            style: .plain,
+            target: self,
+            action: #selector(loadPreviousOrder)
+        )
+
+        // The buttons are too far apart when setting them to rightBarButtonItems, let's adjust the inset to provide better visuals
+        upArrowButon.imageInsets = UIEdgeInsets(top: 0, left: 15, bottom: 0, right: 0)
+        upArrowButon.isEnabled = viewModels[safe: currentIndex - 1] != nil
+
+        let downArrowButon = UIBarButtonItem(
+            image: UIImage(systemName: "chevron.down"),
+            style: .plain,
+            target: self,
+            action: #selector(loadNextOrder)
+        )
+
+        downArrowButon.isEnabled = viewModels[safe: currentIndex + 1] != nil
+
+        return [downArrowButon, upArrowButon]
+    }
+
+    @objc func loadPreviousOrder() {
+        loadOrder(with: currentIndex - 1)
+    }
+
+    @objc func loadNextOrder() {
+        loadOrder(with: currentIndex + 1)
+    }
+
+    func loadOrder(with index: Int) {
+        guard let navigationController = isSplitViewInOrdersTabEnabled ?
+                splitViewController?.viewControllers.first as? UINavigationController : navigationController else {
+            return
+        }
```

Comment:
I find this bit hard to follow, could be more readable with some sort switch statement, or declaring/assigning the variables locally?

Reply:
Sure, I tried to make it more readable in f257f1612901127ec291c105b4086210a3e216ab